cliquet/resource/__init__.py

```diff
@@ -811,6 +829,30 @@ def _raise_400_if_id_mismatch(self, new_id, record_id):
             }
             raise_invalid(self.request, **error_details)
 
+    def _do_projection(self, record, partial_fields):
+        """Project record on partial_fields."""
+
+        return {field: record.get(field) for field in partial_fields}
+
+    def _extract_fields(self):
+        """Extract the fields to do the projection from QueryString parameters.
+        """
+
+        fields = self.request.GET.get('_fields', None)
+        if fields:
+            fields = fields.split(',')
+            known_fields = self._get_known_fields()
+            if set(fields) - set(known_fields):
+                error_msg = "Fields {} do not exist".\
+                    format(set(fields) - set(known_fields))
```

Comment:
nitpick: for better readabilty I suggest that you use an intermediary variable to store the intersection